#2931 phase 1 erisc app FW

aliuTT opened the pull request:
Done:
- erisc FW build on device init
- erisc data transfer protocol, some host driven unit tests passing on N300
- initial eth core L1 memory design
- stripped out open source able tt_eth_api

Major outstanding (P2):
- Program integration somewhat ready, but needs testing
- Load and run reader/writer kernels

pgkeller reviewed (changes requested):
Nice work, that's a lot of stuff to navigate.  Only concern is the tmu-crt0.

aliuTT committed:
#2931: Add initial check-in of erisc application (phase 1)

Build ethernet FW on Device init
Open source tt eth api
Initial spec of eth L1 mem map
Tests to directly send binaries from erisc_app buffer space

TT-billteng reviewed (approved):
Great work!
